io/disk/tiobench.py

```diff
@@ -128,15 +154,146 @@ def create_fs(self, l_disk, mountpoint, fstype):
+        else:
+            self.log.info("No softwareraid detected ")
+        self.log.info("\n End of pre_cleanup")
+
+    def delete_raid(self):
+        """
+        it checks for existing of raid and deletes it if exists
+        """
+        self.log.info("deleting Sraid %s" % self.raid_name)
+
+        def is_raid_deleted():
+            self.sraid.stop()
+            self.sraid.clear_superblock()
+            self.log.info("checking for raid metadata")
+            cmd = "wipefs -af %s" % self.disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if self.sraid.is_raid_exists():
+                return False
+            return True
+        self.log.info("checking lvm_metadata on %s" % self.raid_name)
+        cmd = 'blkid -o value -s TYPE %s' % self.raid_name
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.raid_name
+            process.system(cmd, shell=True, ignore_status=True)
+        if wait.wait_for(is_raid_deleted, timeout=10):
+            self.log.info("software raid  %s deleted" % self.raid_name)
+        else:
+            self.log.info("failed to delete softwareraid %s" % self.raid_name)
+
+    def delete_lv(self):
+        """
+        checks if lv/vg exists and delete them along with its metadata
+        if exists.
+        """
+        def is_lv_deleted():
+            lv_utils.lv_remove(self.vgname, self.lvname)
+            time.sleep(5)
+            lv_utils.vg_remove(self.vgname)
+            if lv_utils.lv_check(self.vgname, self.lvname):
+                return False
+            return True
+        if wait.wait_for(is_lv_deleted, timeout=10):
+            self.log.info("lv %s deleted" % self.lvname)
+        else:
+            self.log.info("lv %s not deleted" % self.lvname)
+        # checking and deleteing if lvm_meta_data exists after lv removed
+        cmd = 'blkid -o value -s TYPE %s' % self.lv_disk
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.lv_disk
+            process.system(cmd, shell=True, ignore_status=True)
+
     def delete_fs(self, l_disk):
-        self.part_obj.unmount()
-        delete_fs = "dd if=/dev/zero bs=512 count=512 of=%s" % l_disk
+        """
+        checks for disk/dir mount, unmount if mounted and checks for
+        filesystem exitance and wipe it off after dir/disk unmount.
+
+        :param l_disk: disk name for which you want to check the mount status
+
+        :return: None
+        """
+        def is_fs_deleted():
+            cmd = "wipefs -af %s" % l_disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if disk.is_fs_exists(l_disk):
+                return False
+            return True
```

Comment:
is_fs_deleted name says you are checking for fs.. but actually deleting fs and than checking?   why not rename to delete_fs ?  also for all below functions too ?

Reply:
@abdhaleegit The return expectation from this function is whether the filesystem deleted properly or not, so it includes delete part too.
Hence I still feel these names looks good to me. if I am missing something to consider in naming part please correct me.
@PraveenPenguin ^^